pkg/plugins/runtime/k8s/webhooks/injector/injector.go

```diff
@@ -455,6 +513,24 @@ func (i *KumaInjector) NewAnnotations(pod *kube_core.Pod, mesh *core_mesh.MeshRe
 		annotations[metadata.CNCFNetworkAnnotation] = metadata.KumaCNI
 	}
 
+	if i.cfg.EBPF.Enabled {
+		// ebpf works only with experimental transparent proxy engine, so instead of
+		// failing when no annotation enabling it is present (bad user experience)
+		// we implicitly add it and set to true
+		enabled, exists, err := metadata.Annotations(pod.Annotations).GetEnabled(metadata.KumaTransparentProxyingExperimentalEngine)
+		if err != nil {
+			return nil, errors.Wrap(err, fmt.Sprintf("getting %s annotation failed", metadata.KumaTransparentProxyingExperimentalEngine))
+		}
+		if !exists || !enabled {
+			log.V(1).Info(fmt.Sprintf("missing %s annotation which has to be %s for ebpf to work. The annotation will be implicitly added",
```

Comment:
Might be worth piping down the logger from above. @jakubdyszkiewicz what's your opinion?

Reply:
addressed